build-tools/automation/build.groovy

```diff
@@ -10,12 +10,24 @@ def isPr = false                // Default to CI
 
 def buildTarget = 'jenkins'
 
-def stageWithTimeout(stageName, timeoutValue, timeoutUnit, directory, fatal, Closure body) {
+def stageWithTimeout(stageName, timeoutValue, timeoutUnit, directory, fatal, ctAttempts, Closure body) {
     try {
         stage(stageName) {
-            timeout(time: timeoutValue, unit: timeoutUnit) {
-                dir(directory) {
-                    body()
+            def retryAttempt = 0
+            def waitSecondsBeforeRetry = 15
```

Comment:
This value may be a bit too high.  Perhaps we should start with 1 or 3 seconds then graduate up based on future results.